addons/agent_mirrorpeer_controller.go

```diff
@@ -417,11 +417,12 @@ func (r *MirrorPeerReconciler) toggleMirroring(ctx context.Context, storageClust
 		return err
 	}
 
+	if sc.Spec.Mirroring == nil {
+		sc.Spec.Mirroring = &ocsv1.MirroringSpec{}
+	}
```

Comment:
We do not want to initialize it everytime.
We only want to initialize it when MCO is the one controlling the enable/disable of mirroring. That was the reason we moved the initialization under the `if enabled {` block.

@rewantsoni Since the mirroring spec is managed by 2 different controllers, I think we need to document how to work with this spec and the expected behavior.